### PR TITLE
dupe-krill: init at 1.4.8

### PR DESCRIPTION
**Pull request by urbas**
###### Description of changes

Added a new Rust application, called dupe-krill at v1.4.8 (see https://github.com/kornelski/dupe-krill for more info).

Unfortunately, dupe-krill does not come with a `Cargo.lock` file, so I had to add it here.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by urbas** [comment]
Result of `nixpkgs-review pr 196298` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dupe-krill</li>
  </ul>
</details>

File: pkgs/tools/filesystems/dupe-krill/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        rev = "v${version}";
+        sha256 = "sha256-DwuUTWm9PdY8bZ+jXa492ibefAlR3FyXEq50ZZITZzs=";
+      };
+
+    in
+    runCommand "dupe-krill-with-cargo.lock" { } ''
+      cp -R ${source} $out
+      chmod +w $out
+      cp ${./Cargo.lock} $out/Cargo.lock
+    '';
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/dupe-krill/default.nix:21]: This can easily be done via postFetch

Comment (urbas) [reply pkgs/tools/filesystems/dupe-krill/default.nix]: Nice! Converted to `postFetch`. :+1:

**Comment by urbas** [comment]
I just created a PR in the dupe-krill repository in hopes to add the `Cargo.lock` file straight to the repo. Here's the PR: https://github.com/kornelski/dupe-krill/pull/15

Apologies for this, but I think it's better if I hold off merging this PR until I hear back from the maintainer of dupe-krill.

**Comment by SuperSandro2000** [comment]
We can drop it with the next release.

**Pull request merged**